### PR TITLE
Remove  deprecated, unreachable_code and add docs to Evm circuit

### PR DESCRIPTION
**Pull request by ChihChengLiang**
### Description

- rm deprecated, unreachable_code
- doc some parts.

### Issue Link

part of  #1494

### Type of change

- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by ChihChengLiang** [comment]
I realized removing `#![allow(missing_docs)]` doesn't enforce documentation in constraint builders.
But it enforces documentation in public structs that were introduced when moving `stats` to `bin`.

**Review by KimiWu123 — APPROVED** [review]
LGTM!

File: zkevm-circuits/src/evm_circuit.rs [hunk 1]
```diff
@@ -183,26 +183,29 @@ impl<F: Field> EvmCircuit<F> {
             fixed_table_tags: FixedTableTag::iter().collect(),
         }
     }
-
-    pub fn get_test_cicuit_from_block(block: Block<F>) -> Self {
+    #[cfg(feature = "test-circuits")]
+    /// Construct the EvmCircuit with only subset of Fixed table tags required by tests to save
+    /// testing time
+    pub(crate) fn get_test_cicuit_from_block(block: Block<F>) -> Self {
```
Comment (hero78119) [inline zkevm-circuits/src/evm_circuit.rs:189]: Here a small typo `cicuit`

**Review by hero78119 — APPROVED** [review]
LGTM! (peeking a small typo :P)

File: zkevm-circuits/src/evm_circuit.rs (hunk 1 above)
Comment (ChihChengLiang) [reply zkevm-circuits/src/evm_circuit.rs]: Nice catch. Fixed in [4277a5b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1524/commits/4277a5bcf40a5fdd660480f76df3f4302ec8ad09)

**Pull request merged**